### PR TITLE
feat: ipython kernel code interpreter

### LINKED ISSUE #692: [Feature Request] Integrate ipython kernel for code execution
### Required prerequisites

- [X] I have searched the [Issue Tracker](https://github.com/camel-ai/camel/issues) and [Discussions](https://github.com/camel-ai/camel/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [ ] Consider asking first in a [Discussion](https://github.com/camel-ai/camel/discussions/new).

### Motivation

more code execution support

### Solution

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by onemquan**
## Description

Added ipython kernel code interpreter

## Motivation and Context

close #692

- [x] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Appointat** [comment]
@onemquan Hi, how are you? Could you please fix the ci/cd?

File: camel/interpreters/ipython_interpreter.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+if TYPE_CHECKING:
+    from jupyter_client import BlockingKernelClient, KernelManager
+
+TIMEOUT = 30
+
+
+def clean_ipython_output(output: str) -> str:
+    ansi_escape = re.compile(r'\x1B[@-_][0-?]*[ -/]*[@-~]')
+    return ansi_escape.sub('', output)
```
Comment (raywhoelse) [inline camel/interpreters/ipython_interpreter.py:30]: it's better to remove to common folder or set it as private func

**Comment by onemquan** [comment]
> @onemquan Hi, how are you? Could you please fix the ci/cd?

the error is caused by `jupyter_client ` not installed in the ci/cd pipeline. but the lib is stated in toml file. Do you have any ideas?

**Review by Appointat — APPROVED** [review]
Hi, @onemquan , regarding to the issue of the dependencies (checked by ci/cd). I have pushed some commits for it.  And thanks for your commits!

**Review by Wendong-Fan — APPROVED** [review]
Thanks @onemquan ! I added one commit to polish docstring and format, please review the commit in below, if it's good for you then we can merge this PR~

https://github.com/camel-ai/camel/pull/726

**Pull request merged**